android/app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreateActivity.kt

```diff
@@ -463,13 +473,46 @@ class CircuitCreateActivity : BaseActivity() {
         }
     }
 
+    private fun navBack(result: Int?) {
+        if (!isTaskRoot) {
+            setResult(result!!)
+            finish()
+        } else {
+            val navToMainActivity = Intent(this, MainActivity::class.java)
+            navToMainActivity.putExtra("deeplinkResult", result)
+            startActivity(navToMainActivity)
+            finish()
+        }
+    }
+
     private fun hideCursors() {
         bind.circuitName.isCursorVisible = false
         bind.setNum.isCursorVisible = false
         bind.setRest.isCursorVisible = false
         bind.setWork.isCursorVisible = false
     }
 
+    private fun handleIntent(intent: Intent) {
+        val intentAction = intent.action
+        val intentData = intent.data
+        if (Intent.ACTION_VIEW == intentAction) {
```

Comment:
Intent.ACTION_VIEW is too generic. In the future, if we use this activity for creating workouts (reps and weight) we would still have an ACTION_VIEW intent but wont be able to distinguish if the ACTION_VIEW intent was intended to create a circuit or workout.